### PR TITLE
KV set improvements for 5.21

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
This is a clean cherry-pick of the various changes to address a number of issues with the plugin key value store.

#### Ticket Links
* https://mattermost.atlassian.net/browse/MM-22074 (https://github.com/mattermost/mattermost-server/pull/13769)
* https://github.com/mattermost/mattermost-server/pull/13753
* https://mattermost.atlassian.net/browse/MM-21672 (https://github.com/mattermost/mattermost-server/pull/13858)
* https://github.com/mattermost/mattermost-server/pull/13943

### CODE REVIEW

**Comment by hanzei** [comment]
@lieut-data Two requests:

- Would you please link the PR that you are consolidating here?
- One tests seam to be failing on Jenkins: https://build.mattermost.com/blue/organizations/jenkins/mp%2Fmattermost-server-pr-new/detail/kv-set-improvements-5.21/1/tests

**Comment by lieut-data** [comment]
Thanks, @hanzei -- debugging the build failure led to https://github.com/mattermost/mattermost-server/pull/13943. I've also updated the PR description with links to the original PRs. Once that's approved and merged, I'll include it in this PR.

File: store/sqlstore/plugin_store.go [hunk 1]
```diff
@@ -50,12 +60,7 @@ func (ps SqlPluginStore) SaveOrUpdate(kv *model.PluginKeyValue) (*model.PluginKe
 		} else if rowsAffected == 0 {
 			// No rows were affected by the update, so let's try an insert
 			if err := ps.GetMaster().Insert(kv); err != nil {
-				// If the error is from unique constraints violation, it's the result of a
-				// valid race and we can report success. Otherwise we have a real error and
-				// need to return it
-				if !IsUniqueConstraintError(err, []string{"PRIMARY", "PluginId", "Key", "PKey"}) {
```
Comment (jespino) [inline store/sqlstore/plugin_store.go]: Is this no longer needed? Why?

Comment (jespino) [reply store/sqlstore/plugin_store.go]: Nevermind, I saw the conversation around the other PR and the final decision of handling this from the client side.

**Comment by lieut-data** [comment]
Skipping tests since is a cherry-pick to the release branch.

**Pull request merged**